README.md

```diff
@@ -400,7 +400,41 @@ The first `stream` argument is the `stream` to use for `rmm::mr::thrust_allocato
 The second `stream` argument is what should be used to execute the Thrust algorithm.
 These two arguments must be identical.
 
-## Debug Logging
+## Logging
+
+RMM includes two forms of logging. Memory event logging and debug logging.
+
+### Memory Event Logging and `logging_resource_adaptor`
+
+Memory event logging writes details of every allocation or deallocation to a CSV (comma-separated
+value) file. In C++, Memory Event Logging is enabled by using the `logging_resource_adaptor` as a 
+wrapper around any other `device_memory_resource` object.
+
+Each row in the log represents either an allocation or a deallocation. The columns of the file are
+"Thread, Time, Action, Pointer, Size, Stream".
+
+The CSV output files of the `logging_resource_adaptor` can be used as input to `REPLAY_BENCHMARK`,
+which is available when building RMM from source, in the `gbenchmarks` folder in the build directory.
+This log replayer can be useful for profiling and debugging allocator issues.
+
+The following C++ example creates a logging version of a `cuda_memory_resource` that outputs the log
+to the file "logs/test1.csv".
+
+```c++
+std::string filename{"logs/test1.csv"};
+rmm::mr::cuda_memory_resource upstream;
+rmm::mr::logging_resource_adaptor<rmm::mr::cuda_memory_resource> log_mr{&upstream, filename};
+```
+
+If a file name is not specified, the environment variable `RMM_LOG_FILE` is queried for the file 
+name. If `RMM_LOG_FILE` is not set, then an exception is thrown by the `logging_resource_adaptor`
+constructor.
+
+In Python, memory event logging is enabled when the `logging` parameter of `rmm.reinitialize()` is
+set to `True`. The log file name can be set using the `log_file_name` parameter. See
+`help(rmm.reinitialize)` for full details.
```

Comment:
We also have the ability for a developer to directly construct a `logging_resource_adaptor` as well but I don't think we want to advertise that interface.